src/mito2/src/worker/handle_write.rs

```diff
@@ -83,12 +94,33 @@ impl<S> RegionWorkerLoop<S> {
                     continue;
                 };
 
-                // Initialize the context.
-                e.insert(RegionWriteCtx::new(region));
+                // A new region to write, checks whether we need to flush this region.
+                self.flush_region_if_full(&region);
+                // Checks whether the region is stalling.
+                let region_ctx_opt = if self.flush_scheduler.is_stalling(region_id) {
+                    // We use `None` to represent the region exists but is stalling so
+                    // there is no write context for it.
+                    None
+                } else {
+                    // Initialize the context.
+                    Some(RegionWriteCtx::new(region))
+                };
+
+                e.insert(region_ctx_opt);
             }
 
             // Safety: Now we ensure the region exists.
-            let region_ctx = region_ctxs.get_mut(&region_id).unwrap();
+            let region_ctx_opt = region_ctxs.get_mut(&region_id).unwrap();
```

Comment:
Not an optional value
```suggestion
            let region_ctx = region_ctxs.get_mut(&region_id).unwrap();
```

Reply:
Replace with an enum in https://github.com/GreptimeTeam/greptimedb/pull/2262/commits/919bc62bcd296992a9098417f867b00e8b072bb6